rest/handler.go

```diff
@@ -267,24 +267,30 @@ func (h *handler) invoke(method handlerMethod, accessPermissions []Permission, r
 	var dbContext *db.DatabaseContext
 	if keyspaceDb != "" {
 		h.addDatabaseLogContext(keyspaceDb)
-		if dbContext, err = h.server.GetDatabase(h.ctx(), keyspaceDb); err != nil {
-			base.InfofCtx(h.ctx(), base.KeyHTTP, "Error trying to get db %s: %v", base.MD(keyspaceDb), err)
-
-			if shouldCheckAdminAuth {
-				if httpError, ok := err.(*base.HTTPError); ok && httpError.Status == http.StatusNotFound {
+		if dbContext, err = h.server.GetActiveDatabase(keyspaceDb); err != nil {
+			base.InfofCtx(h.ctx(), base.KeyHTTP, "Error trying to get active db %s: %v", base.MD(keyspaceDb), err)
+			if base.IsHTTPErrorStatusNotFound(err) {
+				if shouldCheckAdminAuth {
+					// Check if authenticated before attempting to get inactive database
 					authorized, err := h.checkAdminAuthenticationOnly()
 					if err != nil {
 						return err
 					}
-
-					if authorized {
+					if !authorized {
+						return base.HTTPErrorf(http.StatusUnauthorized, "")
+					}
+				}
+				dbContext, err = h.server.GetInactiveDatabase(h.ctx(), keyspaceDb)
+				if err != nil {
+					base.InfofCtx(h.ctx(), base.KeyHTTP, "Error trying to get inactive db %s: %v", base.MD(keyspaceDb), err)
```

Comment:
If this is a not found error, I think that's an expected error case that doesn't require info-level logging.   I think we should only log in the case of an unexpected error.